data/lib/release/changelog_gen.py

```
@@ -0,0 +1,197 @@
+        "",
+        f"# v{semver} 发布说明",
+        zh_body,
+        "",
+        cliff_body,
+    ]
+    zh_path.write_text("\n".join(zh_lines), encoding="utf-8")
+
+
+def _open_editor(filepath: str) -> None:
+    editor = os.environ.get("EDITOR") or os.environ.get("VISUAL") or "vim"
+    subprocess.run([editor, filepath], check=True)
```

Ruff (0.15.15)

[error] `subprocess` call: check for execution of untrusted input

(S603)